--- /Developer/SDKs/MacOSX10.4u.sdk/usr/include/TargetConditionals.h	2006-09-11 21:13:49.000000000 -0500
+++ ./TargetConditionals.h	2007-09-11 11:34:39.000000000 -0500
@@ -26,6 +26,7 @@
         TARGET_CPU_MIPS         - Compiler is generating MIPS instructions
         TARGET_CPU_SPARC        - Compiler is generating Sparc instructions
         TARGET_CPU_ALPHA        - Compiler is generating Dec Alpha instructions
+        TARGET_CPU_ARM          - Compiler is generating ARM instructions
 
 
     TARGET_OS_* 
@@ -67,6 +68,7 @@
         #define TARGET_CPU_MIPS         0
         #define TARGET_CPU_SPARC        0   
         #define TARGET_CPU_ALPHA        0
+        #define TARGET_CPU_ARM          0 
         #define TARGET_RT_LITTLE_ENDIAN 0
         #define TARGET_RT_BIG_ENDIAN    1
         #define TARGET_RT_64_BIT        0
@@ -86,6 +88,7 @@
         #define TARGET_CPU_MIPS         0
         #define TARGET_CPU_SPARC        0   
         #define TARGET_CPU_ALPHA        0
+        #define TARGET_CPU_ARM          0 
         #define TARGET_RT_LITTLE_ENDIAN 0
         #define TARGET_RT_BIG_ENDIAN    1
         #define TARGET_RT_64_BIT        1
@@ -100,6 +103,7 @@
         #define TARGET_CPU_MIPS         0
         #define TARGET_CPU_SPARC        0
         #define TARGET_CPU_ALPHA        0
+        #define TARGET_CPU_ARM          0 
         #define TARGET_RT_MAC_CFM       0
         #define TARGET_RT_MAC_MACHO     1
         #define TARGET_RT_LITTLE_ENDIAN 1
@@ -114,11 +118,27 @@
         #define TARGET_CPU_MIPS         0
         #define TARGET_CPU_SPARC        0
         #define TARGET_CPU_ALPHA        0
+        #define TARGET_CPU_ARM          0 
         #define TARGET_RT_MAC_CFM       0
         #define TARGET_RT_MAC_MACHO     1
         #define TARGET_RT_LITTLE_ENDIAN 1
         #define TARGET_RT_BIG_ENDIAN    0
         #define TARGET_RT_64_BIT        1
+    #elif defined(__ARM__)
+        #define TARGET_CPU_PPC          0
+        #define TARGET_CPU_PPC64        0
+        #define TARGET_CPU_68K          0
+        #define TARGET_CPU_X86          0
+        #define TARGET_CPU_X86_64       0
+        #define TARGET_CPU_MIPS         0
+        #define TARGET_CPU_SPARC        0
+        #define TARGET_CPU_ALPHA        0
+        #define TARGET_CPU_ARM          1 
+        #define TARGET_RT_MAC_CFM       0
+        #define TARGET_RT_MAC_MACHO     1
+        #define TARGET_RT_LITTLE_ENDIAN 1
+        #define TARGET_RT_BIG_ENDIAN    0
+        #define TARGET_RT_64_BIT        0 
     #else
         #error unrecognized GNU C compiler
     #endif
@@ -249,3 +269,4 @@
 #endif
 
 #endif  /* __TARGETCONDITIONALS__ */
+
